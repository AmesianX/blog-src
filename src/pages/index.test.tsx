import React from 'react'
import { useStaticQuery } from 'gatsby'
import { render, cleanup } from '@testing-library/react'

import IndexPage from './index'

jest.mock('hooks/use-site-metadata')

describe('page/IndexPage', () => {
    afterEach(cleanup)

    it('should render properly', () => {
        // @ts-ignore
        useStaticQuery.mockReturnValueOnce({
            allMarkdownRemark: {
                edges: [
                    {
                        node: {
                            frontmatter: {
                                title: '제목',
                                author: '작성자',
                                date: 'YYYY년 MM월 DD일',
                                tags: ['a', 'b', 'c']
                            },
                            fields: {
                                slug: '/slug/',
                                series: '시리즈',
                            },
                            excerpt: '요약',
                        },
                    }
                ],
            },
        })

        const { container } = render(<IndexPage/>)
        expect(container).toMatchSnapshot()
    })
})
