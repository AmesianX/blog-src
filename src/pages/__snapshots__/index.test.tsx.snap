// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page/IndexPage should render properly 1`] = `
.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  width: 100%;
  height: 3.75rem;
  background: #fff;
  border-bottom: 1px solid #e1e1e1;
  -webkit-transition: -webkit-transform .5s ease;
  -webkit-transition: transform .5s ease;
  transition: transform .5s ease;
  z-index: 2;
}

.c0.hide {
  -webkit-transform: translateY(-3.75rem);
  -ms-transform: translateY(-3.75rem);
  transform: translateY(-3.75rem);
}

.c1 {
  position: absolute;
  left: 1.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
  font-weight: bold;
  color: rgba(0,0,0,0.78);
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  width: 100%;
  height: 3rem;
}

.c15 {
  text-align: center;
}

.c15,
.c15 a {
  font-size: .8rem;
  color: darkgray;
}

.c15 + :before {
  content: '·';
  margin: 0 .1rem;
}

.c15 a {
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11 {
  margin: 0;
  padding-left: 0;
  list-style: none;
}

.c13 {
  color: rgba(0,0,0,0.78);
}

.c12 {
  float: left;
  font-size: .75rem;
  background-color: #e1e1e1;
  border-radius: 3px;
  margin: .2rem;
  padding: .15rem .6rem;
}

.c9 {
  margin: .5rem 0;
}

.c10 + :before {
  content: '·';
  margin: 0 .25rem;
}

.c5 {
  overflow: hidden;
  max-width: 50rem;
}

.c6 {
  color: rgba(0,0,0,0.78);
}

.c7 {
  margin-bottom: 0;
}

.c8 {
  font-weight: 200;
}

.c3 {
  list-style: none;
  padding: 0 1.5rem;
}

.c4 {
  padding-bottom: 1.75rem;
  border-bottom: 1px solid #e1e1e1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 3.75rem;
}

<div>
  <header
    class="c0"
  >
    <a
      class="c1"
      href="/"
    >
      cometkim's dev blog
    </a>
  </header>
  <main
    class="c2"
  >
    <ul
      class="c3"
    >
      <li
        class="c4"
      >
        <div
          class="c5"
        >
          <a
            class="c6"
            href="/slug/"
          >
            <h3
              class="c7"
            >
              제목
            </h3>
            <p
              class="c8"
            >
              요약
            </p>
          </a>
          <div
            class="c9"
          >
            <span
              class="c10"
            >
              작성자
            </span>
            <span
              class="c10"
            >
              <time>
                YYYY년 MM월 DD일
              </time>
            </span>
            <span
              class="c10"
            >
              <a
                href="/series/시리즈/"
              >
                시리즈
              </a>
            </span>
          </div>
          <ul
            class="c11"
          >
            <li
              class="c12"
            >
              <a
                class="c13"
                href="/tags/a/"
              >
                a
              </a>
            </li>
            <li
              class="c12"
            >
              <a
                class="c13"
                href="/tags/b/"
              >
                b
              </a>
            </li>
            <li
              class="c12"
            >
              <a
                class="c13"
                href="/tags/c/"
              >
                c
              </a>
            </li>
          </ul>
        </div>
      </li>
    </ul>
  </main>
  <footer
    class="c14"
  >
    <section
      class="c15"
    >
      © 2018 
      <a
        href="/about/"
      >
        Hyeseong Kim
      </a>
    </section>
    <section
      class="c15"
    >
      Powered by 
      <a
        href="https://gatsbyjs.org"
        target="_blank"
      >
        GatsbyJS
      </a>
    </section>
    <section
      class="c15"
    >
      Hosted by 
      <a
        href="https://netlify.com"
        target="_blank"
      >
        Netlify
      </a>
    </section>
    <section
      class="c15"
    >
      Source code on 
      <a
        href="https://github.com/cometkim/blog-src"
        target="_blank"
      >
        GitHub
      </a>
    </section>
  </footer>
</div>
`;
